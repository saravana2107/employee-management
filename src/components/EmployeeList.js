import React from "react";
import Table, { TableColumn } from "./Table";
import { connect } from "react-redux";
import { Button } from "reactstrap";
import { del, toggleModal } from "../actions/employees";

class EmployeeList extends React.Component {
  render() {
    const {
      employees: { data: employees }
    } = this.props;

    const data = [];
    Object.keys(employees).forEach(employee => data.push(employees[employee]));

    return (
      <Table data={data} notFoundText="No Employees Found">
        <TableColumn field="id">ID</TableColumn>
        <TableColumn field="name">Name</TableColumn>
        <TableColumn field="email">Email</TableColumn>
        <TableColumn field="phone">Phone</TableColumn>
        <TableColumn field="address">Address</TableColumn>
        <TableColumn field="city">City</TableColumn>
        <TableColumn field="state">State</TableColumn>
        <TableColumn field="zipcode">ZipCode</TableColumn>
        <TableColumn field="gender">Gender</TableColumn>
        <TableColumn
          field="action"
          renderField={row => (
            <div>
              <Button
                color="primary"
                size="sm"
                className="mr-2"
                onClick={() => this.props.toggleModal(row.id)}
              >
                <i className="fas fa-edit" />
              </Button>
              <Button
                color="danger"
                size="sm"
                onClick={() => this.props.del(row.id)}
              >
                <i className="fas fa-trash" />
              </Button>
            </div>
          )}
        >
          Action
        </TableColumn>
      </Table>
    );
  }
}

export default connect(
  ({ employees }) => ({
    employees
  }),
  { del, toggleModal }
)(EmployeeList);
