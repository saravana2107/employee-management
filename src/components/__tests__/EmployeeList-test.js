import React from "react";
import renderer from "react-test-renderer";
import EmployeeList from "../EmployeeList";
import { Provider } from "react-redux";

function mockStore(state) {
  return {
    default: () => {},
    subscribe: () => {},
    dispatch: () => {},
    getState: () => state,
    mapDispatchToProps: () => {},
    mapStateToProps: () => {}
  };
}

let store;

const employees = {
  modal: false,
  data: {
    ALTI123456: {
      id: "ALTI123456",
      name: "Test User #1",
      email: "test+1@altimetric.com",
      phone: "9876543210",
      address: "Kudlu Gate",
      city: "Bangalore",
      state: "Karnataka",
      zipcode: "560068",
      gender: "Male"
    },
    ALTI123457: {
      id: "ALTI123457",
      name: "Test User #2",
      email: "test+2@altimetric.com",
      phone: "9876543211",
      address: "Kudlu Gate",
      city: "Bangalore",
      state: "Karnataka",
      zipcode: "560068",
      gender: "Male"
    }
  }
};

describe("Employees", () => {
  it("should display employees list", () => {
    store = mockStore({ employees });

    const component = renderer.create(
      <Provider store={store}>
        <EmployeeList />
      </Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
