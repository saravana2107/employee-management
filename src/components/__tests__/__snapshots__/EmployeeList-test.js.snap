// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employees should display employees list 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-striped"
  >
    <thead
      className="thead-light"
    >
      <tr>
        <th>
          ID
        </th>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
        <th>
          Phone
        </th>
        <th>
          Address
        </th>
        <th>
          City
        </th>
        <th>
          State
        </th>
        <th>
          ZipCode
        </th>
        <th>
          Gender
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          className="align-middle"
          scope="row"
        >
          ALTI123456
        </th>
        <td
          className="align-middle"
        >
          Test User #1
        </td>
        <td
          className="align-middle"
        >
          test+1@altimetric.com
        </td>
        <td
          className="align-middle"
        >
          9876543210
        </td>
        <td
          className="align-middle"
        >
          Kudlu Gate
        </td>
        <td
          className="align-middle"
        >
          Bangalore
        </td>
        <td
          className="align-middle"
        >
          Karnataka
        </td>
        <td
          className="align-middle"
        >
          560068
        </td>
        <td
          className="align-middle"
        >
          Male
        </td>
        <td
          className="align-middle"
        >
          <div>
            <button
              aria-label={null}
              className="mr-2 btn btn-primary btn-sm"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-edit"
              />
            </button>
            <button
              aria-label={null}
              className="btn btn-danger btn-sm"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-trash"
              />
            </button>
          </div>
        </td>
      </tr>
      <tr>
        <th
          className="align-middle"
          scope="row"
        >
          ALTI123457
        </th>
        <td
          className="align-middle"
        >
          Test User #2
        </td>
        <td
          className="align-middle"
        >
          test+2@altimetric.com
        </td>
        <td
          className="align-middle"
        >
          9876543211
        </td>
        <td
          className="align-middle"
        >
          Kudlu Gate
        </td>
        <td
          className="align-middle"
        >
          Bangalore
        </td>
        <td
          className="align-middle"
        >
          Karnataka
        </td>
        <td
          className="align-middle"
        >
          560068
        </td>
        <td
          className="align-middle"
        >
          Male
        </td>
        <td
          className="align-middle"
        >
          <div>
            <button
              aria-label={null}
              className="mr-2 btn btn-primary btn-sm"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-edit"
              />
            </button>
            <button
              aria-label={null}
              className="btn btn-danger btn-sm"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-trash"
              />
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
